net/instaweb/rewriter/public/rewrite_options.h

```diff
@@ -4220,6 +4247,8 @@ class RewriteOptions {
   int experiment_percent_;  // Total traffic going through experiments.
   std::vector<ExperimentSpec*> experiment_specs_;
 
+  // Headers to add resource responses.
+  std::vector<NameValue*> resource_headers_;
```

Comment:
@jmarantz 
I'm looking into wrapping RewriteOptions::resource_headers_ in a CopyOnWrite as suggested. The thing is that currently RewriteOptions::resource_headers_ is just a std::vector<NameValue*>, which doesn't have the required Merge() method for CopyOnWrite to work with.

I think it's worth factoring out a class which has the required Merge() method + validations implemented, especially as we can also reuse that for CustomFetchHeaders.

Questions: 
- Does that sound good? 
- If so, do you want me to do so now, or first target merging without wrapping CopyOnWrite, and subsequently refactor / wrap CopyOnWrap in a second pull? A second pull might also be a good place to address a few small things for CustomFetchHeaders (move its system tests to shared, share validations and CopyOnWrap usage with AddResourceHeaders through the new class)?